pkg/storaged/client.js

```diff
@@ -404,6 +407,89 @@
         return utils.compare_versions(this.manager.Version, version) < 0;
     };
 
+    /* NFS mounts
+     */
+
+    function nfs_mounts() {
+        var self = {
+            entries: [ ],
+
+            update_entry: update_entry,
+            add_entry: add_entry,
+            remove_entry: remove_entry,
+
+            entry_users: entry_users,
+
+            mount_entry: mount_entry,
+            unmount_entry: unmount_entry,
+            stop_and_unmount_entry: stop_and_unmount_entry,
+            stop_and_remove_entry: stop_and_remove_entry
+        };
+
+        function spawn_nfs_mounts(args) {
+            return cockpit.spawn(pyinvoke.concat(args),
+                                 { superuser: "try", err: "message" })
+                .input(nfs_mounts_py);
+        }
+
+        var buf = "";
+        spawn_nfs_mounts([ "monitor" ])
+            .stream(function (output) {
+                var lines;
+
+                buf += output;
+                lines = buf.split("\n");
+                buf = lines[lines.length-1];
+                if (lines.length >= 2) {
+                    self.entries = JSON.parse(lines[lines.length-2]);
+                    $(self).triggerHandler('changed');
+                }
+            }).
+            fail(function (error) {
+                if (error != "closed") {
+                    console.warn(error);
+                }
+            });
+
+        function update_entry(entry, new_fields) {
+            return spawn_nfs_mounts([ "update", JSON.stringify(entry), JSON.stringify(new_fields) ]);
+        }
+
+        function add_entry(fields) {
+            return spawn_nfs_mounts([ "add", JSON.stringify(fields) ]);
+        }
+
+        function remove_entry(entry) {
+            return spawn_nfs_mounts([ "remove", JSON.stringify(entry) ]);
+        }
+
+        function mount_entry(entry) {
+            return spawn_nfs_mounts([ "mount", JSON.stringify(entry) ]);
+        }
+
+        function unmount_entry(entry) {
+            return spawn_nfs_mounts([ "unmount", JSON.stringify(entry) ]);
+        }
```

Comment:
This could just call the `mount`/`umount` binaries directly, instead of going through the overhead of the Python execution? That also just calls these through subprocess, and not much else?

Reply:
I think the overhead is small enough to be bearable and that we don't need to make this optimization.  This is not happening in some inner loop, after all. :-)

I prefer it this way for symmetry.  nfs-mounts.py exposes an API, and client.js talks to it.  nfs-mounts.py 'knows' how to call mount, and client.js doesn't need to.  In the future, I hope that udisks2 exposes the API.

Reply:
Also, there is not just "unmount", but also "stop-and-unmount".